### PR TITLE
IMPAC-598 threshold improv

### PR DESCRIPTION
**Pull request by xaun**
TODO: 
- [x] Add loading indicator when the KPI is being added

### CODE REVIEW

File: src/services/kpis/kpis.svc.coffee [hunk 1]
```diff
@@ -341,7 +341,7 @@ angular
               ImpacEvents.notifyCallbacks(IMPAC_EVENTS.addOrRemoveAlerts)
               angular.extend(kpi, success.data)
               _self.buildKpiWatchables(kpi)
-              _self.show(kpi)
+              _self.show(kpi) if showKpi
```
Comment (cesar-tonnoir) [inline src/services/kpis/kpis.svc.coffee]: :+1: indeed...

File: src/services/highcharts-factory/highcharts-factory.svc.coffee [hunk 2]
```diff
@@ -87,16 +82,32 @@ angular
             y: -5
         }]
 
-    thresholdsMarkers: (options = {})->
-      return {} if _.isEmpty(options.thresholds)
-      lines = []
+    addThresholds: (options = @options)->
```
Comment (cesar-tonnoir) [inline src/services/highcharts-factory/highcharts-factory.svc.coffee]: Just out of curiosity: did you try to keep the plotLine, adding a zone with color = transparent before today?
I'm just wondering if we can avoid the fake serie and adding all this code?..

Comment (cesar-tonnoir) [inline src/services/highcharts-factory/highcharts-factory.svc.coffee]: Also, I think that you should pass an argument there to decide whether or not you want to display the threshold only in the future: we may want to display it on the full time period on other widgets

Comment (xaun) [inline src/services/highcharts-factory/highcharts-factory.svc.coffee]: The plotLine doesn't seem to have a Zones API :\ Ok yes I can add an arg for full-length threshold

**Comment by xaun** [comment]
@cesar-tonnoir please review latest commits - thanks

**Comment by xaun** [comment]
https://github.com/maestrano/maestrano-hub/pull/536

**Comment by xaun** [comment]
Yes, there are two linked PRs on this PR. You need mnoe & mno-hub - they have been merged but make sure you have your dev including the required commits.

For the delete / recreate KPI, you mean the series yeah? well you cannot `chart.update()` for new data, when for example you refresh the widget, its best to remove all, and re-add.

For updating the KPI, I guess I could write a new function to `chart.series[0].update()`, but I just thought as the front-end effect would be the same, redrawing the line, that I would re-use the existing update function, rather than adding logic to the factory to determine whether a kpi was being specifically updated. I could look into this further to see if I can find a better way if you think it's worth it.

**Pull request merged**